src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.html

```diff
@@ -25,98 +25,39 @@
-      <li routerLinkActive="active"
-          class="tc_menuitem tc_menuitem_dashboard">
-        <a routerLink="/dashboard">
-          <i class="fa fa-heartbeat fa-fw"
-             [ngStyle]="summaryData?.health_status | healthColor"></i>
-          <span i18n>Dashboard</span>
-        </a>
+      <li>
+        <a><span class="fa fa-heartbeat fa-fw"
+            [ngStyle]="summaryData?.health_status | healthColor"></span></a>
       </li>
 
-      <!-- Cluster -->
-      <li dropdown
-          routerLinkActive="active"
-          class="dropdown tc_menuitem tc_menuitem_cluster"
-          *ngIf="permissions.hosts.read || permissions.monitor.read || permissions.osd.read || permissions.configOpt.read">
-        <a dropdownToggle
-           class="dropdown-toggle"
-           data-toggle="dropdown">
-          <ng-container i18n>Cluster</ng-container>
-          <span class="caret"></span>
-        </a>
-        <ul *dropdownMenu
-            class="dropdown-menu">
-          <li routerLinkActive="active"
-              class="tc_submenuitem tc_submenuitem_hosts"
-              *ngIf="permissions.hosts.read">
-            <a i18n
-               class="dropdown-item"
-               routerLink="/hosts">Hosts</a>
-          </li>
-          <li routerLinkActive="active"
-              class="tc_submenuitem tc_submenuitem_cluster_monitor"
-              *ngIf="permissions.monitor.read">
-            <a i18n
-               class="dropdown-item"
-               routerLink="/monitor/">Monitors</a>
-          </li>
-          <li routerLinkActive="active"
-              class="tc_submenuitem tc_submenuitem_hosts"
-              *ngIf="permissions.osd.read">
-            <a i18n
-               class="dropdown-item"
-               routerLink="/osd">OSDs</a>
-          </li>
-          <li routerLinkActive="active"
-              class="tc_submenuitem tc_submenuitem_configuration"
-              *ngIf="permissions.configOpt.read">
-            <a i18n
-               class="dropdown-item"
-               routerLink="/configuration">Configuration</a>
-          </li>
-          <li routerLinkActive="active"
-              class="tc_submenuitem tc_submenuitem_crush"
-              *ngIf="permissions.hosts.read && permissions.osd.read">
-            <a i18n
-               class="dropdown-item"
-               routerLink="/crush-map">CRUSH map</a>
-          </li>
-          <li routerLinkActive="active"
-              class="tc_submenuitem tc_submenuitem_modules"
-              *ngIf="permissions.configOpt.read">
-            <a i18n
-               class="dropdown-item"
-               routerLink="/mgr-modules">Manager modules</a>
-          </li>
-          <li routerLinkActive="active"
-              class="tc_submenuitem tc_submenuitem_log"
-              *ngIf="permissions.log.read">
-            <a i18n
-               class="dropdown-item"
-               routerLink="/logs">Logs</a>
+      <ng-container *ngFor="let item of menuEntries">
```

Comment:
I'm thinking on moving this to a separate component, and start using onPush change detection strategy, so methods can be invoked directly from the template (since with the Default strategy that would mean invoking a method 11 unnecessarily times per iteration).